### ISSUE TITLE
Release builds can't load cheats

### ISSUE THREAD

**Issue by Epicpkmn11** [ahezard/nds-bootstrap#518]
### Your console

Tell us which console do you have:

- [ ]  DS Phat/Lite
- [x]  DSi (XL)
- [ ]  Old 3DS/2DS (XL)
- [ ]  New 3DS/2DS (XL)

### CFW and configuration

Tell us which CFW do you have:

DSi:

- [ ]  (No CFW) DSiWarehax (such as Flipnote Lenny, Sudokuhax, etc.)
- [ ]  (No CFW) Unlaunch only _[please specify version]_
- [x]  Unlaunch+HiyaCFW _1.8 / 1.3.2_

### nds-bootstrap version

Tell us which versions you are using:

- [x]  Release _0.16.0, 0.17.0_
- [x]  Nightly _69c2a16, 4777e38_

### Expected behavior

Cheats are loaded

### Actual behavior

When I use a release build of 0.16.0 or 0.17.0 and enable cheats on my Japanese DSi with the latest TWiLight, HiyaCFW, and Unlaunch it just gets stuck at a white screen, however if I build it my self on the exact same commit the cheats work fine.

[This](https://github.com/ahezard/nds-bootstrap/files/2820711/nds-bootstrap-release.nds.zip) build of 4777e38 works for me, but the one included with TWiLight v7.3.0 just gets stuck on a white screen, it was the same for v0.16.0.

### Steps to reproduce

1. Enable cheats on a game
2. Use the release bootstrap (or the nightly file included with TWiLight, it's the same thing)
3. Fail
4. Build it yourself (My build above ^)
5. Cheats work fine

### Other notes

My DSi is a Japanese DSi with 1.4.5J and the latest TWiLight, HiyaCFW, and Unlaunch installed.
My computer is a MacBookAir6,2 with macOS 10.14.2 Mojave, I have the latest devkitARM and have used both the latest libnds and Robz's copy of libnds he shared on TWL Mode Hacking a little while ago (that build happens to have been with Robz's copy, but it works for me either way)

**Comment by FM1337** [comment]
I'm having the same issue on the 3DS

**Comment by RocketRobz** [comment]
That's strange. I don't have this issue.
Cheats work fine in GTA Chinatown Wars.

**Comment by FM1337** [comment]
It appears that the following games are having issues

- Pokemon White 2
- Pokemon Black 2

**Comment by Epicpkmn11** [comment]
I had the issue on Pokemon White (J).
I think I tried on a couple other games last release too

**Comment by FM1337** [comment]
@Epicpkmn11 even with your compiled build I still have issues with the games above

**Comment by Epicpkmn11** [comment]
Huh, just tried White 2 and DeadSkullzJr's Walk through walls v2 works fine on my build, maybe try different codes, as some codes make it not boot even with proper builds and all.

**Comment by FM1337** [comment]
I'm wondering: if a code is too long will that stop it from booting?

**Comment by Epicpkmn11** [comment]
I think nds-bootstrap has the same limit as nitrohax, which I think is 2000 characters or something like that... (I forget exactly what it is). I'm not sure what happens if you hit the limit, but I wouldn't be surprised if it was unable to boot

iPhoneから送信

2019/02/01 17:10、"Allen" <notifications@github.com<mailto:notifications@github.com>> のメッセージ:


I'm wondering: if a code is too long will that stop it from booting?

**Comment by FM1337** [comment]
From what I saw on another issue: the issue might be that the area of RAM the cheats are trying to be loaded into, is being used by homebrew.

**Comment by 4AM-Campfire** [comment]
I've run into a similar problem on my new 2DS.

A game (FFTA2) that I had a few cheat enabled worked fine.
I enabled one more and it still worked (an all shop items code).
I then tried to disable the shop items one, but the system won't open the cheat menu any more. It just hangs after you press X and then shuts down.
Trying to launch the game gives a white screen into a shutdown.

Other games play fine, and I can open the cheats menu on them too. It's just FFTA2 that has stopped working.


I'm not sure where the selected cheats data is stored to remove it and try again and see if I have the same problem.
